### PR TITLE
Memo frontend

### PR DESCRIPTION
**Pull request by dnsantosuosso**
A lot of changes:

- Finish Scanner
- Reset password
- Handle all backend responses/loading in UI:
    - Sign Up
    - Sign In
    - Scan
    - Profile
    - Home
    - Purchase ticket
- Validate password length, etc
- Validate Email
- Validate phone number with Flowbite
- Tickets downloadable
- Deleted dup files in pages dir

### CODE REVIEW

File: app/(auth)/reset-password/page.tsx [hunk 1]
```diff
@@ -1,39 +1,76 @@
-export const metadata = {
-  title: 'Reset Password - Simple',
-  description: 'Page description',
-}
+'use client';
```
Comment (chrisjaimes) [inline app/(auth)/reset-password/page.tsx:1]: app/(auth)/password/reset/page.tsx

File: app/Tables/TableOne.tsx [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React, { useEffect, useState } from 'react';
+import Link from 'next/link';
+import { DELETE } from '../api/events/route';
```
Comment (chrisjaimes) [inline app/Tables/TableOne.tsx:3]: Remove

File: app/admin-event/[id]/page.tsx [hunk 3]
```diff
@@ -1,31 +1,95 @@
...
+    if (scanLocked) {
+      console.log('Scan is temporarily locked. Please wait...');
+      return;
     }
-  }, [router.isReady, router.query.id]);
+
+    setScanLocked(true);
+    setTimeout(() => setScanLocked(false), SCAN_TIMEOUT);
+
+    fetch(`/api/tickets/scan`, {
```
Comment (chrisjaimes) [inline app/admin-event/[id]/page.tsx:62]: const response = await fetch

File: app/api/producers/route.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export async function GET(request: Request) {
+  return await fetch(`${process.env.BASE_API_URL}/api/producers`, {
+    headers: {
+      'Content-Type': 'application/json',
```
Comment (chrisjaimes) [inline app/api/producers/route.ts:4]: "Accept-Encoding": "none"

Porque si no vamos a fritar años con Azure como me paso a mi

File: app/api/tickets/route.ts [hunk 5]
```diff
@@ -1,11 +1,11 @@
 export async function GET(request: Request) {
   const { searchParams } = new URL(request.url);
-  const confirmation = searchParams.get("confirmation");
+  const confirmation = searchParams.get('confirmation');
   return await fetch(
     `${process.env.BASE_API_URL}/api/tickets?confirmation=${confirmation}`,
     {
       headers: {
-        "Content-Type": "application/json",
+        'Content-Type': 'application/json',
```
Comment (chrisjaimes) [inline app/api/tickets/route.ts:8]: "Accept-Encoding": "none"

Porque si no vamos a fritar años con Azure como me paso a mi

File: app/api/tickets/scan/route.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  console.log('We are inside this fetch for scanning');
+  console.log(body.ticketId);
+  console.log(process.env.BASE_API_URL);
+  console.log(`${process.env.BASE_API_URL}/api/tickets/scan/${body.ticketId}`);
+
+  // Assuming the backend endpoint is generic and ticketId should not be a path parameter
+  return fetch(`${process.env.BASE_API_URL}/api/tickets/scan`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
```
Comment (chrisjaimes) [inline app/api/tickets/scan/route.ts:12]: "Accept-Encoding": "none"

Porque si no vamos a fritar años con Azure como me paso a mi

File: app/api/tickets/scan/route.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
+export async function POST(request: Request) {
+  const body = await request.json();
+  console.log('We are inside this fetch for scanning');
```
Comment (chrisjaimes) [inline app/api/tickets/scan/route.ts:3]: quitar console.log

File: app/event/[id]/page.tsx [hunk 8]
```diff
@@ -518,8 +544,8 @@ const EventDetailPage = ({ params }: { params: { id: string } }) => {
                   type="button"
                   className={`btn-sm text-white ml-3 ${
                     Object.values(ticketTypesQty).every((qty) => qty == 0)
-                      ? "bg-gray-600"
-                      : "bg-green-600 hover:bg-green-700"
+                      ? 'bg-gray-600'
+                      : 'bg-green-600 hover:bg-green-700'
```
Comment (chrisjaimes) [inline app/event/[id]/page.tsx:548]: disabled={Object.values(ticketTypesQty).every(
    (qty) => qty == 0
) || isCheckingOut}

File: app/payment/success/[confirmation]/page.tsx [hunk 9]
```diff
@@ -1,5 +1,20 @@
-"use client";
-import React, { useState, useEffect } from "react";
+'use client';
+import React, { useState, useEffect } from 'react';
+import { jsPDF } from 'jspdf';
+import { format } from 'date-fns';
+
+// Define TypeScript interfaces for clarity and type safety
```
Comment (chrisjaimes) [inline app/payment/success/[confirmation]/page.tsx:6]: Remove both interface Ticket and Event and replace with:

import { Event } from "@/app/models/Event";
import { Ticket } from "@/app/models/Ticket";

asi no definimos las clases siempre

File: app/Tables/TableOne.tsx (hunk 2 above)
Comment (dnsantosuosso) [reply app/Tables/TableOne.tsx:3]: JAJAJAJA

File: app/event/[id]/page.tsx (hunk 8 above)
Comment (dnsantosuosso) [reply app/event/[id]/page.tsx:548]: Buena

File: app/payment/success/[confirmation]/page.tsx (hunk 9 above)
Comment (dnsantosuosso) [reply app/payment/success/[confirmation]/page.tsx:6]: Buena

File: app/admin-event/[id]/page.tsx (hunk 3 above)
Comment (dnsantosuosso) [reply app/admin-event/[id]/page.tsx:62]: Bro es que si hago eso no estaria propagando el message del server

**Review by chrisjaimes — APPROVED** [review]

**Pull request merged**